### PR TITLE
Resolve environment variables in environmentPath and executablePath settings

### PR DESCRIPTION
**Pull request by coloursofnoise**
Partial implementation of #1781 and #1135.

Uses the [VSCode Variable](https://code.visualstudio.com/docs/editor/variables-reference#_environment-variables) format to resolve environment variables in the `environmentPath` setting before it is used.

Since there doesn't currently seem to be a [public API for VSCode variables](https://github.com/microsoft/vscode/issues/46471), resolving *all* variables could potentially be done though the [vscode-variables](https://github.com/DominicVonk/vscode-variables) package, although it seems to be in need of work since it includes the [deprecated `vscode` package](https://github.com/DominicVonk/vscode-variables/pull/1).

### CODE REVIEW

File: src/jlpkgenv.ts [hunk 1]
```diff
@@ -196,6 +202,12 @@ function absEnvPath(p: string) {
     }
 }
 
+function parseEnvVariables(p: string) {
+    return p.replace(/\${env:(.*?)}/g, function (variable) {
+        return process.env[variable.match(/\${env:(.*?)}/)[1]] || ''
```
Comment (pfitzseb) [inline src/jlpkgenv.ts:207]:
```suggestion
    return p.replace(/\${env:(.*?)}/g, (_, variable) => {
        return process.env[variable] || ''
```
No need to apply the regex twice.

**Comment by pfitzseb** [comment]
I like it. Makes sense for both env path and executable path.

**Comment by coloursofnoise** [comment]
I'm not entirely sure how to merge upstream changes without including submodule updates...

**Comment by pfitzseb** [comment]
`git merge master` shouldn't pull in any submodule updates, IIRC.

**Comment by coloursofnoise** [comment]
@pfitzseb is there anything preventing this from getting merged?

**Review by pfitzseb — APPROVED** [review]

**Comment by pfitzseb** [comment]
Yep, me forgetting to press "merge" :) Thanks for the reminder.

**Comment by bradcarman** [comment]
Note: this doesn't appear to be working for me.  I'm on Windows 11 and running VS Code 1.103.1 with Julia Extension v1.149.2

[https://github.com/julia-vscode/julia-vscode/issues/1135#issuecomment-3211307899](https://github.com/julia-vscode/julia-vscode/issues/1135#issuecomment-3211307899)

**Pull request merged**